buffer/src/main/java/io/netty/buffer/api/CompositeBuffer.java

```diff
@@ -791,6 +803,32 @@ public CompositeBuffer splitComponentsCeil(int splitOffset) {
         return buildSplitBuffer(splits);
     }
 
+    /**
+     * Break a composite buffer into its constituent components.
+     * <p>
+     * This "consumes" the composite buffer, leaving the composite buffer instance as if it had been closed.
+     * The buffers in the returned array are not closed, and become owned by the caller.
+     *
+     * @return An array of the constituent buffer components.
+     */
+    public Buffer[] decomposeBuffer() {
+        Buffer[] result = bufs;
+        bufs = EMPTY_BUFFER_ARRAY;
+        try {
+            close();
```

Comment:
Also I wonder if what you do here is really what is expected. Like if `decomposeBuffer` throws would you expect it to have a side-effect or not. When I did add a comment I was more thinking of something like this:

```
public Buffer[] decomposeBuffer() {
     Buffer[] result = bufs;
     bufs = EMPTY_BUFFER_ARRAY;
     try {
         close();
     } catch (Throwable cause) {
         bufs = result;
         throw cause;
     }
}
```

Reply:
It depends on why it throws. If we made a transition from not-closed to closed, then the buffer cannot be closed again, and re-inserting the buffer array will make them leak.